### ISSUE TITLE
arangorestore crashes with a large 'indexBuckets' size

### ISSUE THREAD

**Issue by iamacup** [arangodb/arangodb#1708]
Hey - I ran a dump of arango running on my local machine with this command:

`arangodump --include-system-collections false --output-directory "dump7" --dump-data false`

I ran this command on the other machine

`arangorestore --input-directory "dump7"`

I got this output:

```
# ... a bunch of successful imports
# Re-creating edge collection 'at_company'...
# Re-creating edge collection 'default_followon_question'...
Error: got invalid response from server: Could not connect to 'tcp://127.0.0.1:8529' 'connect() failed with #111 - Connection refused'
```

I ran this with a few different combinations of outputs (with, without data and system collections) with no success - always failing on that command - eventually i found that the problem was the indexBuckets value - here is the JSON that is inside of default_followon_question structure.json file

` {"parameters":{"version":5,"type":3,"cid":"74120776865","indexBuckets":301228032,"deleted":false,"doCompact":true,"maximalSize":33554432,"name":"default_followon_question","isVolatile":false,"waitForSync":false},"indexes":[]}`

i looked in all of the other structure.json files, the indexBuckets value was 8 for all of them. I changed the value in this specific file to 8, and the whole import ran without a problem.

- I don't know what indexBuckets does - it might be reasonable for the DB to say no to 300 million god damn buckets?
- I didn't set it to this massive value  - if it should not be this high then something else is wrong?
- I think the import should fail with a better error like 'indexBuckets is huge you nutter' instead of connection error, if indeed it should fail at all on massive indexBuckets

I have a side question - will i suffer any ill side effects of having indexBuckets manually set to 8 on this collection?

both machines are running 2.8.1

**Comment by jsteemann** [comment]
Setting `indexBuckets` to 8 manually will do no harm, so your workaround was just right.
The error was due to the indexBuckets not being initialized properly in one special case. In this case its value was unpredicatable (301228032 in your case). 

The issue is fixed in the 2.7, 2.8 and devel branch now, and the `indexBuckets` size is sanitized to be between 1 and 1024.

**Comment by dothebart** [comment]
the 2.8.x releases with the fixes are available for download.
